tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs

```diff
@@ -248,6 +249,36 @@ public async Task AndroidMessageHandlerFollows308PermanentRedirect ()
 			Assert.AreEqual ("https://www.microsoft.com/", result.RequestMessage.RequestUri.ToString ());
 		}
 
+		[Test]
+		public async Task AndroidMessageHandlerSendsClientCertificate ([Values(true, false)] bool setClientCertificateOptionsExplicitly)
+		{
+			using X509Certificate2 certificate = BuildClientCertificate ();
+
+			using var handler = new AndroidMessageHandler ();
+			if (setClientCertificateOptionsExplicitly) {
+				handler.ClientCertificateOptions = ClientCertificateOption.Manual;
+			}
+			handler.ClientCertificates.Add (certificate);
+
+			using var client = new HttpClient (handler);
+			var response = await client.GetAsync ("https://corefx-net-tls.azurewebsites.net/EchoClientCertificate.ashx");
+			var content = await response.EnsureSuccessStatusCode ().Content.ReadAsStringAsync ();
+
+			X509Certificate2 certificate2 = new X509Certificate2 (global::System.Convert.FromBase64String (content));
+			Assert.AreEqual (certificate.Thumbprint, certificate2.Thumbprint);
+		}
+
+		[Test]
+		public async Task AndroidMessageHandlerRejectsClientCertificateOptionsAutomatic ()
```

Comment:
This test concerns me, as it's (kinda, sorta) the "inverse" of [my previous comment](https://github.com/xamarin/xamarin-android/pull/8961#discussion_r1610537993).  This test asserts that when `AndroidMessageHandler.ClientCertificateOptions` is `ClientCertificateOption.Automatic`, accessing `AndroidMessageHandler.ClientCertificates` will throw.

Fair enough.

But what should happen when `ClientCertificateOptions` is set *last*?

```csharp
var h = new AndroidMessageHandler {
    ClientCertificates = { BuildClientCertificate () },
    ClientCertificateOptions = ClientCertificateOption.Automatic,
};
var c = new HttpClient(h);
var r = await c.GetAsync (…);
```

*Because* you've updated the default value of `AndroidMessageHandler.ClientCertificateOptions` to `ClientCertificateOption.Manual`, I *believe* that `h` will be constructed without error, because *when `h.ClientCertificates` is accessed*, `ClientCertificateOptions` will still be `.Manual`, so `h.ClientCertificates` shouldn't throw.

The result is that *after construction*, `h` is in an invalid state.

What will happen when we try to use `h`?  What *should* happen?

Reply:
> What will happen when you try to use `h`? What should happen?

1. When we set `ClientCertificateOptions.Automatic` first and then we set `ClientCertificates`, we know we're in an invalid state as soon as we set `ClientCertificates` (they would be ignored). HttpClientHandler throws which breaks the design guideline that we should allow being temporarily in an invalid state.
    - I think we _should_ follow the design of HttpClientHandler to keep the behavior consistent accross platforms since `AndroidMessageHandler` is commonly used through `HttpClientHandler`.

2. In the other scenario, when we set `ClientCertificates` first and then we set `ClientCertificateOptions.Automatic`, the HttpClientHandler/SocketsHttpHandler is not in an invalid state because it chooses to ignore the `ClientCertificates` completely (certificates from the current user's X509 store are used instead).
    - We could maybe support the Automatic mode as well (using KeyChain.choosePrivateKeyAlias to show the user a UI to choose a certificate?) but definitely not in this PR. Until we support the `Automatic` mode, I think it makes sense to throw once we are preparing the request here: https://github.com/xamarin/xamarin-android/pull/8961/files?diff=split&w=0#diff-5db393c58158459e46590be1423607c52488da9182e9e5e929d7b79e7179d13dR1219-R1221 Maybe this deserves its own unit test.
    - Alternatively, we could choose to throw as early as when the developer sets `ClientCertificateOptions.Automatic` because we don't support it at all. If we want to go in this direction, we should coordinate this with the iOS/Mac implementation (cc @dotMorten)

Reply:
That makes sense: `ClientCertificateOptions = ClientCertificateOption.Automatic` means "*ignore* `ClientCertificates`", so the resulting state *isn't* "invalid", it's just "weird" (contains data that cannot be used until `ClientCertificateOptions` is changed again).

@simonrozsival: what's the process for getting documentation updated?  The documentation for [`HttpClientHandler.ClientCertificates`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.clientcertificates?view=net-8.0) does not mention that it could throw an exception depending on the value of [`HttpClientHandler.ClientCertificateOptions`](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.clientcertificateoptions?view=net-8.0).